stdlib/asm/collections/smt.masm

```diff
@@ -0,0 +1,82 @@
+#! input:  [v, ...]
+#! output: [v >> 48, ...]
+proc.shr_48
+    u32split
+    push.65536
+    mul
+    u32split
+    swap
+    drop
+    swap
+    drop
+end
+
+#! input:  [v, ...]
+#! output: [v >> 32, ...]
+proc.shr_32
+    u32split
+    swap
+    drop
+end
+
+#! input:  [v, ...]
+#! output: [v >> 16, ...]
+proc.shr_16
+    u32split
+    push.65536
+    mul
+    u32split
+    push.4294967296
+    mul
+    swap.2
+    push.65536
+    mul
+    u32split
+    swap
+    drop
+    add
+    add
+end
+
+#! Fetches the keyed value of a Sparse Merkle tree with the provided root.
+#! cycles: 91
+#! input:  [K, R, ...]
+#! output: [d, i, V, K, R, ...]
+export.get
+    # invoke the advice injector and fetch depth+index pair
+    # (18 cycles)
+    # [d, i, R, d, K, i, d, ...]
+    adv.smtget swapw adv_push.2 dup movdn.6 dup.1 movdn.11 dup movdn.12
```

Comment:
We can do that, but that will add cost to the operation. It's worthy to note that the advice provider index output is verified by `mtree_get`, so it is guaranteed to be consistent

Reply:
The advice provider could give any other index at the same depth and `mtree_get` would still work. For non-empty leaves it will be caught when we check the pre-image of the node, but for empty leaves it will cause a problem.

Reply:
What we check is `mtree_get` with `[d, i, H_d(K, V), R, ...]`; it won't need to open the pre-image as just this opening is enough to check non-empty leaves. For empty leaves, indeed this `smt::get` is not prepared to handle that. Let's continue the discussion here:

https://github.com/0xPolygonMiden/miden-vm/pull/808#discussion_r1154010287